src/Application/Common/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+namespace Application.Common.Interfaces
+{
+    public interface IRepository<T> where T : class
+    {
+        void SaveChanges();
+
+        Task SaveChangesAsync();
+
+        IQueryable<T> GetAll();
```

Comment:
Repository shouldn't return IQueryable<T>. It should either IEnumerable<T> or List<T>